services/core/shared/database/knex.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { Logger } = require('lisk-service-framework');
+
+const logger = Logger();
+
+const createDb = async () => {
+    const knex = require('knex')({
+        client: 'mysql',
+        version: '5.7',
+        connection: {
+            host: '127.0.0.1',
+            user: 'lisk',
+            password: 'password',
+            database: 'lisk',
+        },
+        // useNullAsDefault: true,
+        log: {
+            warn(message) { logger.warn(message); },
+            error(message) { logger.error(message); },
+            deprecate(message) { logger.deprecate(message); },
+            debug(message) { logger.debug(message); },
+        },
+        migrations: {
+            directory: './database/knex_migrations',
+            loadExtensions: ['.js'],
+        },
+    });
+
+    await knex.migrate.latest();
+
+    return knex;
+};
+
+let dbConnection = null;
+
+const getDbInstance = async (tableName) => {
+    if (!dbConnection) dbConnection = await createDb();
+
+    const knex = dbConnection;
+
+    const write = async (row) => knex.transaction(async trx => {
+        const inserts = await trx(tableName).insert(row).onConflict('id').merge()
```

Comment:
This needs a discussion - if the `write()` method overwrites data then there is a need for `update()`.
Now `write()` does updates by default.

Reply:
#296